File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string sentence = get_string("Enter text: ");
    int letter = count_letters(sentence);
    // printf("%i letters \n", letter);
    int word = count_words(sentence);
    // printf("%i words \n", word);
    int sentences = count_sentences(sentence);
    // printf("%i sentences \n", sentences);
    float index = 0.;
    float l = 100. * letter / word;
    float s = 100. * sentences / word;
    index = 0.0588 * l - 0.296 * s - 15.8;
    // printf("%f is index \n", index);
    // printf("L= %f \n", l);
    // printf("S= %f \n", s);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %g\n", round(index));
    }
}

int count_letters(string text)
{
    int letter = 0;
    char c;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        c = text[i];
        if (isalpha(c))
        {
            letter++;
        }
    }
    return letter;
}

int count_words(string text)
{
    int word = 0;
    char c;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        c = text[i];
        if (c == ' ')
        {
            word++;
        }
    }
    word++;
    return word;
}

int count_sentences(string text)
{
    int sentence = 0;
    char c;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        c = text[i];
        if (c == '.' || c == '!' || c == '?')
        {
            sentence++;
        }
    }
    return sentence;
}

File: readability.py
# TODO
from cs50 import get_string


def count_letters(text):
    letter = 0
    for i in text:
        print(i)
        if i.isalpha():
            letter += 1
    return letter


def count_words(text):
    word = 0
    for i in text:
        if i == " ":
            word += 1
    word += 1
    return word


def count_sentences(text):
    sentence = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            sentence += 1
    return sentence


sentence = get_string("Enter text: ")
letter = count_letters(sentence)
print(f"{letter} letters")
word = count_words(sentence)
print(f"{word} words")
sentences = count_sentences(sentence)
print(f"{sentences} sentences")
index = 0.0
l = 100.0 * letter / word
s = 100.0 * sentences / word
index = 0.0588 * l - 0.296 * s - 15.8
print(f"{index} is index")
print(f"L= {l}")
print(f"S= {s}")
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")
